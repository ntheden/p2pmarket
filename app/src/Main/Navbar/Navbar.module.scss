.navbar {
    background-color: white;
    border-bottom: 1px solid rgba(var(--b6a, 219, 219, 219), 1);

    .brand {
        display: flex;
        align-content: center;
    }
}
.searchInput {
    background-color: white;
}

.links {
    display: flex;
    justify-content: space-between;
    svg {
        width: 1.25em;
        height: 1.25em;
    }
}
