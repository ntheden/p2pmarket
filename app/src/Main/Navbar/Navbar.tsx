import React, { useEffect, useState } from "react";
import { Sling as Hamburger } from 'hamburger-react'
import {
    BsSearch,
    BsHouseDoor,
    BsMessenger,
    BsPlusSquare,
    BsCompass,
    BsHeart,
    BsFillEmojiSmileUpsideDownFill,
} from 'react-icons/bs';
import styles from './Navbar.module.scss';

import {
    Col,
    Container,
    FormControl,
    InputGroup,
    Nav,
    Navbar,
    Row,
} from 'react-bootstrap';

export const NavbarLayout = () => {
    const [burgerIsOpen, setBurgerIsOpen] = useState<boolean>(false);

    useEffect(() => {
        if (burgerIsOpen) {
            console.log("Burger is open");
        } else {
            console.log("Burger is closed");
        }
    }, [burgerIsOpen]);

    return (
        <Navbar className={styles.navbar}>
            <Col md={{ offset: 2, span: 9 }}>
                <Container>
                    <Row>
                        <Col md={{ span: 3 }}>
                            <Navbar.Brand href="#home" className={styles.brand}>
                                @bitcoinp2pmarketplace
                            </Navbar.Brand>
                        </Col>
                        <Col md={{ span: 5 }} className={styles.searchBar}>
                            <InputGroup>
                                <InputGroup.Text className={styles.searchInput}>
                                    <BsSearch />
                                </InputGroup.Text>
                                <FormControl
                                    placeholder="Search"
                                    aria-label="Search"
                                />
                            </InputGroup>
                        </Col>
                        <Col md={{ offset: 2, span: 2 }}>
                            <Hamburger toggled={burgerIsOpen} toggle={setBurgerIsOpen}/>
                        </Col>
                    </Row>
                </Container>
            </Col>
        </Navbar>
    )
};
