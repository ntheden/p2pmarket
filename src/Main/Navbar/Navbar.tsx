import {
    BsSearch,
    BsHouseDoor,
    BsMessenger,
    BsPlusSquare,
    BsCompass,
    BsHeart,
    BsFillEmojiSmileUpsideDownFill,
} from 'react-icons/bs';
import styles from './Navbar.module.scss';

import {
    Col,
    Container,
    FormControl,
    InputGroup,
    Nav,
    Navbar,
    Row,
} from 'react-bootstrap';

export const NavbarLayout = () => (
    <Navbar className={styles.navbar}>
        <Col md={{ offset: 2, span: 9 }}>
            <Container>
                <Row>
                    <Col md={{ span: 2 }}>
                        <Navbar.Brand href="#home" className={styles.brand}>
                            P2P Market
                        </Navbar.Brand>
                    </Col>
                    <Col md={{ span: 6 }} className={styles.searchBar}>
                        <InputGroup>
                            <InputGroup.Text className={styles.searchInput}>
                                <BsSearch />
                            </InputGroup.Text>
                            <FormControl
                                placeholder="Search"
                                aria-label="Search"
                            />
                        </InputGroup>
                    </Col>
                    <Col md={{ span: 4 }}>
                        <Nav className={styles.links}>
                            <Nav.Link href="#home">
                                <BsHouseDoor />
                            </Nav.Link>
                            <Nav.Link href="#Inbox">
                                <BsMessenger />
                            </Nav.Link>
                            <Nav.Link href="#NewPost">
                                <BsPlusSquare />
                            </Nav.Link>
                            <Nav.Link href="#Discover">
                                <BsCompass />
                            </Nav.Link>
                            <Nav.Link href="#Notifications">
                                <BsHeart />
                            </Nav.Link>
                            <Nav.Link href="#Notifications">
                                <BsFillEmojiSmileUpsideDownFill />
                            </Nav.Link>
                        </Nav>
                    </Col>
                </Row>
            </Container>
        </Col>
    </Navbar>
);
