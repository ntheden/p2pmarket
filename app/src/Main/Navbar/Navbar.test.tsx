import React from 'react';
import { render, screen } from '@testing-library/react';
import { NavbarLayout } from './Navbar';

test('renders Navbar', () => {
    const { container, getByText, getAllByRole, getByPlaceholderText } = render(
        <NavbarLayout />,
    );

    expect(getByText('P2P Market')).toBeInTheDocument();
    expect(getByPlaceholderText('Search')).toBeInTheDocument();
    expect(getAllByRole('link').length).toEqual(7);

    expect(container).toMatchSnapshot();
});
