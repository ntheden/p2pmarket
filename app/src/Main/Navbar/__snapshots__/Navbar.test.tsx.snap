// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar 1`] = `
<div>
  <nav
    class="navbar navbar navbar-expand navbar-light"
  >
    <div
      class="col-md-9 offset-md-2"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-md-2"
          >
            <a
              class="brand navbar-brand"
              href="#home"
            >
              P2P Market
            </a>
          </div>
          <div
            class="searchBar col-md-6"
          >
            <div
              class="input-group"
            >
              <span
                class="searchInput input-group-text"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
                  />
                </svg>
              </span>
              <input
                aria-label="Search"
                class="form-control"
                placeholder="Search"
              />
            </div>
          </div>
          <div
            class="col-md-4"
          >
            <div
              class="links navbar-nav"
            >
              <a
                class="nav-link"
                data-rr-ui-event-key="#home"
                href="#home"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8.354 1.146a.5.5 0 0 0-.708 0l-6 6A.5.5 0 0 0 1.5 7.5v7a.5.5 0 0 0 .5.5h4.5a.5.5 0 0 0 .5-.5v-4h2v4a.5.5 0 0 0 .5.5H14a.5.5 0 0 0 .5-.5v-7a.5.5 0 0 0-.146-.354L13 5.793V2.5a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5v1.293L8.354 1.146zM2.5 14V7.707l5.5-5.5 5.5 5.5V14H10v-4a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5v4H2.5z"
                  />
                </svg>
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#Inbox"
                href="#Inbox"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 7.76C0 3.301 3.493 0 8 0s8 3.301 8 7.76-3.493 7.76-8 7.76c-.81 0-1.586-.107-2.316-.307a.639.639 0 0 0-.427.03l-1.588.702a.64.64 0 0 1-.898-.566l-.044-1.423a.639.639 0 0 0-.215-.456C.956 12.108 0 10.092 0 7.76zm5.546-1.459-2.35 3.728c-.225.358.214.761.551.506l2.525-1.916a.48.48 0 0 1 .578-.002l1.869 1.402a1.2 1.2 0 0 0 1.735-.32l2.35-3.728c.226-.358-.214-.761-.551-.506L9.728 7.381a.48.48 0 0 1-.578.002L7.281 5.98a1.2 1.2 0 0 0-1.735.32z"
                  />
                </svg>
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#NewPost"
                href="#NewPost"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
                  />
                  <path
                    d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
                  />
                </svg>
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#Discover"
                href="#Discover"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 16.016a7.5 7.5 0 0 0 1.962-14.74A1 1 0 0 0 9 0H7a1 1 0 0 0-.962 1.276A7.5 7.5 0 0 0 8 16.016zm6.5-7.5a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0z"
                  />
                  <path
                    d="m6.94 7.44 4.95-2.83-2.83 4.95-4.949 2.83 2.828-4.95z"
                  />
                </svg>
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#Notifications"
                href="#Notifications"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m8 2.748-.717-.737C5.6.281 2.514.878 1.4 3.053c-.523 1.023-.641 2.5.314 4.385.92 1.815 2.834 3.989 6.286 6.357 3.452-2.368 5.365-4.542 6.286-6.357.955-1.886.838-3.362.314-4.385C13.486.878 10.4.28 8.717 2.01L8 2.748zM8 15C-7.333 4.868 3.279-3.04 7.824 1.143c.06.055.119.112.176.171a3.12 3.12 0 0 1 .176-.17C12.72-3.042 23.333 4.867 8 15z"
                  />
                </svg>
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#Notifications"
                href="#Notifications"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM7 9.5C7 8.672 6.552 8 6 8s-1 .672-1 1.5.448 1.5 1 1.5 1-.672 1-1.5zM4.285 6.433a.5.5 0 0 0 .683-.183A3.498 3.498 0 0 1 8 4.5c1.295 0 2.426.703 3.032 1.75a.5.5 0 0 0 .866-.5A4.498 4.498 0 0 0 8 3.5a4.5 4.5 0 0 0-3.898 2.25.5.5 0 0 0 .183.683zM10 8c-.552 0-1 .672-1 1.5s.448 1.5 1 1.5 1-.672 1-1.5S10.552 8 10 8z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
